components/core/tests/test-NetworkReader.cpp

```diff
@@ -64,6 +72,17 @@ auto get_content(clp::ReaderInterface& reader, size_t read_buf_size) -> std::vec
     }
     return buf;
 }
+
+auto assert_curl_error_code(CURLcode expected, CURLcode actual) -> void {
+    if (expected == actual) {
+        return;
+    }
+    std::string const error_message{
+            "Unexpected CURL error code: " + std::to_string(actual)
+            + "; expected: " + std::to_string(expected)
+    };
+    FAIL(error_message);
```

Comment:
Based on https://curl.se/libcurl/c/libcurl-errors.html, can we also use [CURLOPT_ERRORBUFFER](https://curl.se/libcurl/c/CURLOPT_ERRORBUFFER.html) so that we can get more details about the error (e.g., what specific HTTP error >= 400 it was)?